src/Responses/Responses/Tool/RemoteMcpTool.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenAI\Responses\Responses\Tool;
+
+use OpenAI\Contracts\ResponseContract;
+use OpenAI\Responses\Concerns\ArrayAccessible;
+use OpenAI\Testing\Responses\Concerns\Fakeable;
+
+/**
+ * @phpstan-type RemoteMcpToolType array{type: 'mcp', server_label: string, server_url: string, require_approval: 'never'|null, allowed_tools: string[]|null, headers: array<string, string>|null}
+ *
+ * @implements ResponseContract<RemoteMcpToolType>
+ */
+final class RemoteMcpTool implements ResponseContract
+{
+    /**
+     * @use ArrayAccessible<RemoteMcpToolType>
+     */
+    use ArrayAccessible;
+
+    use Fakeable;
+
+    /**
+     * @param  'mcp'  $type
+     * @param  'never'|null  $requireApproval
```

Comment:
This appears to be an object (ApprovalFilter) which has keys of `always`|`never` of the tools that require approval or not or a setting (`always` | `never`)